src/devices/Ht16k33/SevenSegment.cs

```diff
@@ -0,0 +1,235 @@
+            }
+        }
+        #endregion
+
+        #region Public members
+        /// <summary>
+        /// Initialize 7-Segment display
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        public SevenSegment(I2cDevice i2cDevice) : base(i2cDevice) { }
+
+        /// <summary>
+        /// Gets or sets colon
+        /// </summary>
+        /// <remarks>
+        /// Setting the dot is like a write operation, it doesn't
+        /// automatically flush the buffer to the device, you still
+        /// have to call the Flush method to do that
+        /// </remarks>
+        public Dot Dots
+        {
+            get => (Dot)displayBuffer[(int)Address.Dots];
+            set => displayBuffer[(int)Address.Dots] = (byte)value;
+        }
+
+        /// <summary>
+        /// Gets or sets a single digit's segments by address
+        /// </summary>
+        /// <param name="address">address of digit</param>
+        /// <returns></returns>
+        public byte this[Address address]
+        {
+            get
+            {
+                EnsureValidAddress(address);
+                return displayBuffer[(int)address];
+            }
+            set
+            {
+                EnsureValidAddress(address);
+                displayBuffer[(int)address] = value;
+            }
+        }
+
+
+        /// <summary>
+        /// Write 4 raw digits to display (from left to right)
+        /// </summary>
+        /// <param name="first">first digit</param>
+        /// <param name="second">second digit</param>
+        /// <param name="third">third digit</param>
+        /// <param name="fourth">fourth digit</param>
+        public void Write(byte first, byte second, byte third, byte fourth)
+        {
+            displayBuffer[(int)Address.Digit1] = first;
+            displayBuffer[(int)Address.Digit2] = second;
+            displayBuffer[(int)Address.Digit3] = third;
+            displayBuffer[(int)Address.Digit4] = fourth;
+        }
+
+        /// <summary>
+        /// Write 4 raw digits to display (from left to right)
+        /// </summary>
+        /// <param name="first">first digit</param>
+        /// <param name="second">second digit</param>
+        /// <param name="third">third digit</param>
+        /// <param name="fourth">fourth digit</param>
+        public void Write(Segment first, Segment second, Segment third, Segment fourth) =>
+            Write((byte)first, (byte)second, (byte)third, (byte)fourth);
+
+        /// <summary>
+        /// Write 4 hexadecimal digits display (from left to right)
+        /// </summary>
+        /// <param name="first">first digit</param>
+        /// <param name="second">second digit</param>
+        /// <param name="third">third digit</param>
+        /// <param name="fourth">fourth digit</param>
+        public void WriteHex(byte first, byte second, byte third, byte fourth)
```

Comment:
instead of this and next overload should there be just 1 which takes single `uint`?

Reply:
Good idea, added more overloads.